lib/matplotlib/axes/_base.py

```diff
@@ -2227,41 +2227,58 @@ def margins(self, *margins, x=None, y=None, tight=True):
+        Specifying any margin changes only the autoscaling; for example,
+        if *xmargin* is not None, then *xmargin* times the X data
+        interval will be added to each end of that interval before
+        it is used in autoscaling.
 
-            margins(x=xmargin, y=ymargin)
+        Parameters
+        ----------
+        args : float, optional
+            If a single positional argument is provided, it specifies
+            both margins of the x-axis and y-axis limits. If two
+            positional arguments are provided, they will be interpreted
+            as *xmargin*, *ymargin*. If setting the margin on a single
+            axis is desired, use the keyword arguments described below.
+
+        x, y : float, optional
+            Specific margin values for the x-axis and y-axis,
+            respectively. These cannot be used with positional
+            arguments, but can be used individually to alter on e.g.,
+            only the y-axis.
+
+        tight : bool, default is True
+            The *tight* parameter is passed to :meth:`autoscale_view`,
+            which is executed after a margin is changed; the default
+            here is *True*, on the assumption that when margins are
+            specified, no additional padding to match tick marks is
+            usually desired.  Set *tight* to *None* will preserve
+            the previous setting.
 
-            margins(..., tight=False)
 
-        All three forms above set the xmargin and ymargin parameters.
-        All keyword parameters are optional.  A single positional argument
-        specifies both xmargin and ymargin. The padding added to the end of
-        each interval is *margin* times the data interval. The *margin* must
-        be a float in the range [0, 1].  Passing both positional and keyword
-        arguments for xmargin and/or ymargin is invalid.
+        Returns
+        -------
+        xmargin, ymargin : float
 
-        The *tight* parameter is passed to :meth:`autoscale_view`
-        , which is executed after a margin is changed; the default here is
-        *True*, on the assumption that when margins are specified, no
-        additional padding to match tick marks is usually desired.  Setting
-        *tight* to *None* will preserve the previous setting.
+        Notes
+        -----
+        If a previously used Axes method such as :meth:`pcolor` has set
+        :attr:`use_sticky_edges` to `True`, only the limits not set by
+        the "sticky artists" will be modified. To force all of the
+        margins to be set, set :attr:`use_sticky_edges` to `False`
+        before calling :meth:`margins`.
 
-        Specifying any margin changes only the autoscaling; for example,
-        if *xmargin* is not None, then *xmargin* times the X data
-        interval will be added to each end of that interval before
-        it is used in autoscaling.
+        .. plot:: gallery/subplots_axes_and_figures/axes_margins.py
```

Comment:
@NelleV Should I remove this? I think (but I'm not sure) that with the addition of the References section in the gallery example. we 're getting redundant references to the example:
https://10812-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/api/_as_gen/matplotlib.axes.Axes.margins.html#matplotlib.axes.Axes.margins

Reply:
Yeah, I would now that sphinx-gallery deals with this pretty nicely, it's redundant. Nice catch: I hadn't looked at the circle artifact.